common/flatpak-dir.c

```diff
@@ -2456,10 +2457,19 @@ flatpak_dir_mark_changed (FlatpakDir *self,
                           GError    **error)
 {
   g_autoptr(GFile) changed_file = NULL;
+  g_autofree char * changed_path = NULL;
 
   changed_file = flatpak_dir_get_changed_path (self);
+  changed_path = g_file_get_path (changed_file);
+
+  if (!g_utime (changed_path, NULL))
```

Comment:
Doesn't look right to me. The docs say: "Returns 0 if the operation was successful, -1 otherwise"
In particular, it will return unsuccessful if errno is ENOENT, so checking for ENOENT later won't do much good...

Reply:
So if the utime() call is successful, this is !0, so we will early return TRUE from the function, and the rest of it is the error / ENOENT handling. Previously it was more like:
```
  if (g_utime (..))
    return glnx_throw_errno (error);
```

So the error case was an early return, but we didn't handle the ENOENT case. I did it this way round to avoid the need to nest/indent all of the error handling.

Reply:
More coffee needed on this side, it seems

Reply:
Story of my entire day, I'm glad it wasn't just me. https://github.com/flatpak/flatpak/pull/2162/commits/bb4928aa :)